src/generateTypeFiles.js

```diff
@@ -82,12 +76,51 @@ export const generateTypeFiles = (
                 ? `// To regenerate, run '${options.regenerateCommand}'.\n`
                 : '') +
             code;
+        if (options.splitTypes) {
+            fileContents +=
+                `\nexport type ${name} = ${typeName}['response'];\n` +
+                `export type ${name}Variables = ${typeName}['variables'];\n`;
+        }
 
-        fs.writeFileSync(targetPath, fileContents);
         addToIndex(targetPath, typeName);
+        files[targetPath] = fileContents;
     });
 
+    return {files, indexContents};
+};
+
+export const generateTypeFiles = (
+    fileName: string,
+    schema: Schema,
+    document: DocumentNode,
+    options: Options,
+) => {
+    const generatedDir = path.join(
+        path.dirname(fileName),
+        options.generatedDirectory ?? '__generated__',
+    );
+    const indexFile = path.join(generatedDir, 'index.js');
+
+    if (!fs.existsSync(generatedDir)) {
+        fs.mkdirSync(generatedDir, {recursive: true});
+
+        // Now write an index.js for each __generated__ dir.
+        fs.writeFileSync(indexFile, indexPrelude(options.regenerateCommand));
+    }
+
+    const {indexContents, files} = generateTypeFileContents(
+        fileName,
+        schema,
+        document,
+        options,
+        generatedDir,
+        fs.readFileSync(indexFile, 'utf8'),
+    );
```

Comment:
Good call separating coding that doesn't do any file operations from code that does.